### PR TITLE
refactor(web): simplifies the gesture source paradigm 🐵

### PR DESCRIPTION
**Pull request by jahorton**
While it's hard to succinctly prove exactly _why_ the changes in this PR are necessary, these changes have arisen while working on the next layer for gesture-recognition (which I'm currently terming either "gesture selection" or "gesture staging").  Each issue described below was complicating matters, and these changes:

1. allow work on the next aspects of the gesture feature to proceed more smoothly
2. ... with far fewer complications / "here be dragons" left behind within said "next aspects"
3. and actually simplify a noted unclear aspect of the existing design, likely making it somewhat _more_ intuitive in most parts of the gesture engine.
    - While `ComplexGestureSource` here would have helped simplify data organization for the new layer if viewed in isolation... the next costs - the logic using it - would have outweighed the benefits.
    - I believe this also notably simplifies a significant part of the module's to-be-API surface, so... that's a win.

If you feel like that's too much of a TL;DR... well, see the "Issues from existing design" section for more precise details about what motivated this changeset.

Either way, the complications were great enough that the simplest way for me to push past them... was to make this PR.  Which, fortunately, could easily be spun off from the work that motivated them.

----

### Notes for Reviewers

"Hide whitespace" is your friend for this one.  Especially due to one commit: https://github.com/keymanapp/keyman/commit/7f37a3f92b8368be081d78e767ed6a37a17922bd.  It's solely dedicated to fixing the test fixtures.

1. This is the commit responsible for the lion's share of the changes.
2. Almost all of it is whitespace changes.

The other two commits shouldn't be affected much by "hide whitespace", so feel free to review it either way.

Only the first commit affects the main section of the gesture-engine.

----

#### Complicating assumption

While working on the "gesture selection" layer, it's slowly become evident that the assumptions behind the `ComplexGestureSource` are, at a minimum, unnecessary and can cause significant complications - potential control-flow paths that would have to be "cordoned off" with no easy future resolution.

The assumption:  that we can always know immediately, upon the `touchstart` of an incoming contact point, whether that contact point will be "bound" to another, pre-existing contact point (or set thereof) as part of a multitouch gesture.  

The logic ended up boiling down to:

```
if(any multitouch gesture can accept the new contact-point / gesture source) {
  // then it MUST belong to a multitouch gesture, even if it none instantly resolve upon accepting it
} else {
  // then it's a standard, single contact-point gesture.
}
```

If, at any point, it turns out that no multitouch gestures can be matched with it: "sorry, out of luck."

For the gestures we plan to implement... this assumption, on its own, actually is pretty valid.  

#### Complicating gesture-spec requirement

Secondly, there's the detail that certain gestures we _do_ plan to support have cases where we need to reset the path - but only for matching / recognizing _that specific gesture_.  (Say, the base key has shifted under a longpress that hasn't triggered its subkey menu.  _If_ we wanted to recognize flicks simultaneously, their version of the path should _not_ be reset.)  

Again, for 17.0 we were not going to specify gesture-specs that would allow these two cases to occur at the same time.

**The crux of the matter**: when the two aspects above _collide_, things get very, _very_ messy.  Should one potential two-touch gesture-matcher need a path reset while other still-active two-touch gesture-matchers do _not_... the presence of `ComplexGestureSource` heavily overcomplicates the logic needed to handle such cases.

Preserving things as they were would have practically required flagrant code-comment warnings about "These cases are not handled and will not be easy to handle as is" and things we'd just have to know... not to try.  

Considering it didn't take me too long to find a way around it now, I figured "better to do this while the area's still fresh" rather than have to figure the solution out again at a far later time.

### Better design clarity

Finally, revisiting #9261 and what motivated it... I think it's safe to say that things become a lot simpler if we don't need to distinguish between a `ComplexGestureSource` and a `SimpleGestureSource`.  Having just `GestureSource` seems way more straightforward.  I'm just not making _that_ nomenclature change here because the other edits are already substantial; there's already enough detail to review with the structural changes already happening.

Note that each pre-recorded gesture-path fixture is now internally "simpler" - there's less "digging" needed to get to the important parts - and simpler to _use_ - noting the simplifications to test resource scripts.  Yay for simpler maintenance in those areas!

----

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts [hunk 1]
```diff
@@ -26,26 +25,26 @@ export class GestureMatcher<Type> {
...
     return this.publishedPromise.corePromise;
   }
 
-  constructor(model: GestureModel<Type>, sourceObj: ComplexGestureSource<Type> | GestureMatcher<Type>) {
+  constructor(model: GestureModel<Type>, sourceObj: SimpleGestureSource<Type> | GestureMatcher<Type>) {
     /* c8 ignore next 5 */
     if(!model || !sourceObj) {
       throw new Error("Construction of GestureMatcher requires a gesture-model spec and a source for related contact points.");
-    } else if(!model.sustainTimer && sourceObj instanceof ComplexGestureSource && sourceObj.touchpoints.length == 0) {
+    } else if(!model.sustainTimer && !(sourceObj)) {
```
Comment (mcdurdin) [inline common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts:38]:
```suggestion
    } else if(!model.sustainTimer && !sourceObj) {
```

File: common/web/gesture-recognizer/src/engine/headless/touchpointCoordinator.ts [hunk 2]
```diff
@@ -36,59 +34,42 @@ export class TouchpointCoordinator<HoveredItemType> extends EventEmitter<EventMa
     this.inputEngines.push(engine);
   }
 
-  private readonly onNewTrackedPath = (touchpoint: SimpleGestureSource<HoveredItemType>) => {
+  private readonly onNewTrackedPath = async (touchpoint: SimpleGestureSource<HoveredItemType>) => {
```
Comment (mcdurdin) [inline common/web/gesture-recognizer/src/engine/headless/touchpointCoordinator.ts:37]: why async now? Can you add a return type (`Promise<void>` I guess?) for clarity?

File: common/web/gesture-recognizer/src/test/auto/browser/cases/canary.js [hunk 3]
```diff
@@ -82,10 +82,14 @@ describe("'Canary' checks", function() {
       this.controller.recognizer.on('inputstart', fakeHandler)
       fireEvent();
 
+      // because of the mild optimization await when spinning up gesture-matching
```
Comment (mcdurdin) [inline common/web/gesture-recognizer/src/test/auto/browser/cases/canary.js:85]: What is a "mild optimization await"? Can you clarify?

**Review by mcdurdin — APPROVED** [review]
LGTM. Fewer abstractions and simpler code is a definite positive. Look forward to the nice and tidy renaming PR for `SimpleGestureSource` to `GestureSource`!

File: common/web/gesture-recognizer/src/engine/headless/touchpointCoordinator.ts (hunk 2 above)
Comment (jahorton) [reply common/web/gesture-recognizer/src/engine/headless/touchpointCoordinator.ts:37]: That snuck in from the original changeset I spun this PR off from.  Should definitely revert.

```suggestion
  private readonly onNewTrackedPath = (touchpoint: SimpleGestureSource<HoveredItemType>) => {
```

File: common/web/gesture-recognizer/src/test/auto/browser/cases/canary.js (hunk 3 above)
Comment (jahorton) [reply common/web/gesture-recognizer/src/test/auto/browser/cases/canary.js:85]: Right, another thing that snuck in from the original changeset pre-spinoff.  Will fix.

**Pull request merged**